### PR TITLE
nate/fix/LOOP-1698/insulin-suspended-temp-status-bar

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1698

re-adding insulin suspended temp status bar with cell background colour matching the status bar

### CODE REVIEW

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -588,6 +588,7 @@ final class StatusTableViewController: LoopChartsTableViewController {
         case enactingBolus
         case bolusing(dose: DoseEntry)
         case cancelingBolus
+        case pumpSuspended(resuming: Bool)
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift:591]: [nit] Why not have two separate states for this: `pumpSuspended` and `pumpResuming`?

File: Loop/View Controllers/StatusTableViewController.swift [hunk 2]
```diff
@@ -608,6 +609,10 @@ final class StatusTableViewController: LoopChartsTableViewController {
             statusRowMode = .enactingBolus
         } else if case .canceling = bolusState {
             statusRowMode = .cancelingBolus
+        } else if case .suspended = basalDeliveryState {
+            statusRowMode = .pumpSuspended(resuming: false)
+        } else if self.basalDeliveryState == .resuming {
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift]: [nit] Why the different comparison for this line and the line 2 above? Suggest changing this line to:
```
} else if case .resuming = basalDeliveryState {
```

File: Loop/View Controllers/StatusTableViewController.swift [hunk 3]
```diff
@@ -990,6 +1015,24 @@ final class StatusTableViewController: LoopChartsTableViewController {
...
+                case .pumpSuspended(let resuming) where !resuming:
+                    self.updateHUDandStatusRows(statusRowMode: .pumpSuspended(resuming: true) , newSize: nil, animated: true)
+                    self.deviceManager.pumpManager?.resumeDelivery() { (error) in
+                        DispatchQueue.main.async {
+                            if let error = error {
+                                let alert = UIAlertController(with: error, title: NSLocalizedString("Error Resuming", comment: "The alert title for a resume error"))
+                                self.present(alert, animated: true, completion: nil)
+                                if case .suspended = self.basalDeliveryState {
+                                    self.updateHUDandStatusRows(statusRowMode: .pumpSuspended(resuming: false), newSize: nil, animated: true)
+                                }
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift:1027]: [nit] Do w need to call `refreshContext.update` or `reloadData` here?

**Review by darinkrauss — APPROVED** [review]
I think splitting out `pumpSuspended(resuming: Bool)` into separate states `pumpSuspended` and `pumpResuming` makes sense. What do you think? Otherwise, a couple of nits/questions, but LGTM! 🐳

File: Loop/View Controllers/StatusTableViewController.swift (hunk 1 above)
Comment (nhamming) [reply Loop/View Controllers/StatusTableViewController.swift:591]: So, I'm just re-adding code that already existed and was removed.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**